From bc0fa2826359617bb5e4762f8dc8985ff8cc3861 Mon Sep 17 00:00:00 2001
From: David Mitchell <davem@iabyn.com>
Date: Tue, 30 Jul 2013 20:14:37 +0100
Subject: Disable new //p tests

The previous commit fixed a regression in 5.18.0:

    my $pat = qr/a/;
    'aaaa' =~ /$pat/gp or die;
    print "MATCH=[${^MATCH}]\n";

which printed 'a' in 5.16.0, and undef in 5.18.0.
5.18.0 only broke the /g behaviour; the non-/g variant was already broken
and the previous commit didn't fix that for maint.

The new tests are testing for the non-/g variant, which still fail, so
disable these tests.

(Getting it to capture under non-/g works in 5.19.2 due to a major
reorganisation of the pp_match/pp_subst/regexec() code, which isn't
suitable for backporting to maint-5.18.)

Origin: upstream, http://perl5.git.perl.org/perl.git/commit/4d7b2f562978b8efa58e6a489ac3a0f146794184
Bug: https://rt.perl.org/rt3/Public/Bug/Display.html?id=118213
Bug-Debian: Bug-Debian: http://bugs.debian.org/718209
Patch-Name: fixes/regexp-preserve-testcases.diff
---
 t/re/reg_pmod.t |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/t/re/reg_pmod.t b/t/re/reg_pmod.t
index 1db0bed..106c0dc 100644
--- a/t/re/reg_pmod.t
+++ b/t/re/reg_pmod.t
@@ -12,7 +12,8 @@ use warnings;
 our @tests = (
     # /p      Pattern   PRE     MATCH   POST
     [ '/p',   "345",    "012-", "345",  "-6789"],
-    [ '/$r/p',"345",    "012-", "345",  "-6789"],
+    # these not supported under 5.18.x
+    #[ '/$r/p',"345",    "012-", "345",  "-6789"],
     [ '(?p)', "345",    "012-", "345",  "-6789"],
     [ '(?p:)',"345",    "012-", "345",  "-6789"],
     [ '',     "(345)",  undef,  undef,  undef ],
